sdk/eventhub/azure-eventhubs/azure/eventhub/aio/eventprocessor/event_processor.py

```diff
@@ -205,96 +100,175 @@ def _context(self, events):
+                        log.warning(
+                            "EventProcessor instance %r of eventhub %r partition %r consumer group %r"
+                            " has an error during running initialize(). The exception is %r.",
+                            owner_id, eventhub_name, partition_id, consumer_group_name, err
+                        )
+                while True:
+                    try:
+                        events = await partition_consumer.receive()
+                        self._last_enqueued_event_properties[partition_id] = \
+                            partition_consumer.last_enqueued_event_properties
+                        with self._context(events):
+                            await self._event_handler(partition_context, events)
+                    except asyncio.CancelledError:
+                        log.info(
+                            "EventProcessor instance %r of eventhub %r partition %r consumer group %r"
+                            " is cancelled",
+                            owner_id,
+                            eventhub_name,
+                            partition_id,
+                            consumer_group_name
+                        )
+                        raise
+                    except Exception as error:  # pylint:disable=broad-except
+                        await process_error(error)
+                        break
+                    # Go to finally to stop this partition processor.
+                    # Later an EventProcessor(this one or another one) will pick up this partition again
+            finally:
+                await partition_consumer.close()
+                if self._running is False:
+                    await close(CloseReason.SHUTDOWN)
+                else:
                     await close(CloseReason.OWNERSHIP_LOST)
-                    break
-                except Exception as other_error:  # pylint:disable=broad-except
-                    await process_error(other_error)
-                    await close(CloseReason.PROCESS_EVENTS_ERROR)
-                    break
         finally:
-            await partition_consumer.close()
+            if partition_id in self._tasks:
+                del self._tasks[partition_id]
+
+    async def start(self):
+        """Start the EventProcessor.
+
+        The EventProcessor will try to claim and balance partition ownership with other `EventProcessor`
+         and asynchronously start receiving EventData from EventHub and processing events.
+
+        :return: None
+
+        """
+        log.info("EventProcessor %r is being started", self._id)
+        ownership_manager = OwnershipManager(self._eventhub_client, self._consumer_group_name, self._id,
+                                             self._partition_manager, self._ownership_timeout, self._partition_id)
+        if not self._running:
+            self._running = True
+            while self._running:
```

Comment:
This function is called "start" but it looks like this while loop will remain running for the entire lifetime?

Reply:
Yes, it balances load, cancels tasks for lost partitions and creates tasks for newly gained partitions